### ISSUE TITLE
Twisted doesn't validate HTTP versions

### ISSUE THREAD

**Issue by kenballus** [twisted/twisted#12233]
# Description

Twisted does no validation of received requests' HTTP versions.

# How to cause this behavior

1. Start a Twisted HTTP server that echos the received version, such as [this one](https://github.com/narfindustries/http-garden/blob/main/images/twisted/server.py).
2. Send Twisted an HTTP request with an invalid version, and extract the parsed version from the response:
```sh
(printf 'GET / This-is-totally-invalid!-It-even-contains-unprintable-bytes,-like-\x00-and\xff!!\r\nHost: a\r\n\r\n'; sleep 1) \
    | nc localhost 80 \
    | awk '{RS=""; FS="\r\n\r\n"} {print $2}' \
    | tail -n 1 \
    | jq '.["version"]' \
    | xargs echo \
    | base64 -d \
    | xxd
```
3. Observe that the version is unchanged, including the NUL and non-ASCII byte:
```
00000000: 5468 6973 2d69 732d 746f 7461 6c6c 792d  This-is-totally-
00000010: 696e 7661 6c69 6421 2d49 742d 6576 656e  invalid!-It-even
00000020: 2d63 6f6e 7461 696e 732d 756e 7072 696e  -contains-unprin
00000030: 7461 626c 652d 6279 7465 732c 2d6c 696b  table-bytes,-lik
00000040: 652d 002d 616e 64ff 2121                 e-.-and.!!
```

# The correct behavior
RFC 9112 specifies that HTTP versions have to match the following grammar rule:
```
  HTTP-version  = HTTP-name "/" DIGIT "." DIGIT
  HTTP-name     = %s"HTTP"
```
In other words, HTTP versions need to match the following regex: `HTTP/[0-9]\.[0-9]`.

**Testing environment**
 - Operating System and Version; paste the output of these commands:
   - `Linux e153e0bbe66c 6.9.7-arch1-1 #1 SMP PREEMPT_DYNAMIC Fri, 28 Jun 2024 04:32:50 +0000 x86_64 GNU/Linux`
 - Twisted version [e.g. 22.2.0]
   - `24.3.0.post0`

**Comment by glyph** [comment]
Presumably the correct behavior here would be to issue a 400 response?

**Comment by adiroiban** [comment]
I think that 400 is appropriate

```
$ telnet google.com 80
Trying 142.250.200.110...
Connected to google.com.
Escape character is '^]'.
GET / This-is-totally-invalid!
Host: a

HTTP/1.0 400 Bad Request
```

**Comment by twm** [comment]
Thank you for the report @kenballus! I have addressed this in #12233 by validating that the version is either `HTTP/1.0` or `HTTP/1.1`. My thinking is that all later versions use different framing, so why be lax? Please let me know if you're aware of any compatibility issues that may cause.
